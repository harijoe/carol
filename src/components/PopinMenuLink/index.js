import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { theme } from 'utils/style'

import Link from 'components/Link'

const StyledLink = styled(Link)`
  padding: ${theme('spaces.s')} 0;
  color: ${theme('colors.grayscale.darker')};
  width: 100%;
  display: block;
`

const PopinMenuLink = ({ children, to }) =>
  <StyledLink to={to}>
    {children}
  </StyledLink>

PopinMenuLink.propTypes = {
  children: PropTypes.any.isRequired,
  to: PropTypes.string.isRequired,
}

export default PopinMenuLink
