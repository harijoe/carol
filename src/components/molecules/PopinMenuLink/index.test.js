import React from 'react'
import { shallow } from 'enzyme'

import PopinMenuLink from './'

it('renders PopinMenuLink', () => {
  const wrapper = shallow(<PopinMenuLink to="test">Test</PopinMenuLink>)

  expect(wrapper).toMatchSnapshot()
})
