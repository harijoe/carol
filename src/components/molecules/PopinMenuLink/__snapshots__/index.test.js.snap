// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PopinMenuLink 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PopinMenuLink
    to="test"
  >
    Test
  </PopinMenuLink>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": "Test",
      "to": "test",
    },
    "ref": null,
    "rendered": "Test",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": "Test",
        "to": "test",
      },
      "ref": null,
      "rendered": "Test",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
